clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -26,20 +27,40 @@ const (
 type Config struct {
 	SyncPeriod time.Duration `mapstructure:"sync_period" yaml:"sync_period"`
 	Stdin      bool          `mapstructure:"stdin" yaml:"stdin"`
+	Watch      WatchConfig   `mapstructure:"watch_config" yaml:"watch_config"`
```

Comment:
Move this to a new `global` block, but call it `file_watch_config` to make it clearer that it only applies to file_sd.

Reply:
Actually, this is way harder than it seems. file_sd-specific settings get lost by the time a file target is made. Either it would need to be set at the job level or pushed as a Grafana Agent Flow feature. 

I'd like to make this global-only for now and we can see if people ask for more granular control.